### PR TITLE
`sage.graphs`: Add `# optional` doctest tags for modularization

### PR DESCRIPTION
**Pull request by mkoeppe**
### 📚 Description

These tags make it possible/meaningful to doctest `sage.graphs` even when some "standard" packages (`networkx`, `numpy`, the modularized distribution packaging providing `sage.groups`) are not installed. 

- Part of #29705 
- Split out from #34998

### 📝 Checklist

- [x] I have made sure that the title is self-explanatory and the description concisely explains the PR.
- [x] I have linked an issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### ⌛ Dependencies

This is just one commit on top of:
- #35237

### CODE REVIEW

File: src/sage/graphs/generators/families.py [hunk 1]
```diff
@@ -3409,29 +3409,29 @@ def WheelGraph(n):
 
     Next, using the spring-layout algorithm::
 
-        sage: import networkx
+        sage: import networkx                                                   # optional - sage.groups
```
Comment (dcoudert) [inline src/sage/graphs/generators/families.py]: Should be networkx and not sage.groups

Comment (dimpase) [inline src/sage/graphs/generators/families.py]: did you mean ` - networkx` here?

**Review by dimpase — CHANGES_REQUESTED** [review]
please check that `sage.groups`  tags are not placed by mistake, instead of  `networkx`

**Comment by mkoeppe** [comment]
I've rebased on top of the dependency https://github.com/sagemath/sage/pull/35237

**Review by dimpase — APPROVED** [review]
lgtm

**Review by dcoudert — APPROVED** [review]
LGTM.

File: src/sage/graphs/generators/families.py (hunk 1 above)
Comment (mkoeppe) [reply src/sage/graphs/generators/families.py]: Fixed already, thanks!

**Comment by mkoeppe** [comment]
Thanks!

**Comment by vbraun** [comment]
```
sage -t --long --warn-long 73.8 --random-seed=123 src/sage/interfaces/r.py  # 3 doctests failed
```

**Pull request merged**